### ISSUE TITLE
Optionally require email confirmation

### ISSUE THREAD

**Issue by rixx** [pretalx/pretalx#846]
Currently, submitter and organiser emails are in no way verified, and it's time to change that. Plan:

- [ ] Track if an email address is verified
- [ ] Allow administrators to make email verification to be required on their instance
- [ ] Allow event organisers to restrict email verification settings further for their event (but not relax them, if there are instance settings)

While we're at it, we should also verify sender/reply-to addresses that organisers configure, though that may need to be its own issue.

I'm currently imagining settings like

- off
- delayed (ie deactivate after x days unverified, with x maybe even configurable)
- verify before activating account

Further thoughts:
- emails should only be sent to confirmed/verified email addresses
- emails should only be sent BY users with confirmed/verified email addresses
- how does email confirmation look in the cfp flow? before being able to send in a proposal? does it block saving as draft? etc
- how does the migration process work

**Comment by johanvdw** [comment]
We had a discussion about this, and I think the proposal is too complex which makes it hard to implement.

By not requiring registration people can use the submission form to send spam, any links in the abstract will actually be formatted nicely.

In that view, making validation optional is not really useful, and it seems a better idea to require any email address to be validated on registration.

**Comment by rixx** [comment]
So your counter-proposal would be that all email addresses/ accounts always require verification. This doesn't make it any less complicated, it just moves the complication:

- how do you deal with preexisting accounts?
- which actions do you permit unverified accounts?
- e.g., if verification becomes mandatory during the proposal flow, then unverified accounts already have at least their proposal draft attached – or do you also disallow this?
- would your proposal mean making sign-in/sign-up the *first* step in the proposal flow? we have so far avoided this because people are more likely to submit proposals if they aren't greeted with a signup mask from the get-go …

FWIW, I don't think the settings suggested in this issue have to be implemented immediately – but I also don't think that they are complex at all. I think the complexity is in the UI and the questions I listed above, and that once general verification has been implemented, the additional settings are mostly trivial.

**Comment by primeos** [comment]
Tbh I'm quite surprised that this IMO essential feature still isn't a thing and that it wasn't implemented in the first place.

As an administrator I really don't like to host a service where anyone can trigger an email to any valid email address and control part of the content (including the ability to insert arbitrary links).
But apart from that it ended up creating another issue for us when we (https://www.tuebix.org/) switched to Pretalx last year: A few people made typos in their email addresses and we couldn't contact them (unless we could find the correct address on the web, etc.). A second field to confirm the email address could help here but what we really want is to validate it - that way it's also guaranteed that the submitter can (or at least could) be reached. Since Pretalx doesn't notice if emails failed to deliver this is particularly important.

As for this proposal being too complex: IMO it should be sufficient to provide a setting to enforce email address validation (it could even be the default without a setting as I don't really see good reasons not to have this validation but I guess a setting would be better for "backward compatibility"). Ignoring preexisting accounts should be fine, I think (they basically were fine-ish in the past and admins could also delete them manually). People would then basically create an account first before being able to submit/do anything. @rixx mentioned a valid concern that such a signup mask might "discourage" some people from submitting proposals but I don't feel like this should be a problem in practice since they already have to create an account with a password anyway (and should expect this) and also have to click the ACK link in a mail once their proposal got accepted (so I'd rather have them "fail" to confirm their email address than to fail at this stage by, e.g., not reading their mails).

@rixx what do you think - could this be implemented in such a way?

**Comment by primeos** [comment]
Oh, and the email address confirmation would also be very helpful to avoid "trolling" issues.

Strangers could submit "weird" talks in the name of someone else and this might get noticed too late if they also put a typo into the email address (or even use their own email domain with a fake name).

And if a "troll" creates an account for you with your email and submits anything you'll likely get "spammed" a bit when the proposal gets reviewed, etc. And the worst thing is that you have to do the work to unsubscribe, i.e., reset the account password via email, login, and then delete your account (which I just noticed might not even be possible? if so that also be problematic in regards of the GDPR).

**Comment by rixx** [comment]
> Tbh I'm quite surprised that this IMO essential feature still isn't a thing and that it wasn't implemented in the first place.

I agree, it’s quite interesting to see which issues receive PRs and which do not. Funny how Open Source works!

(If your surprise is aimed at me: I’m afraid as a solo dev who was until very very recently running this project as a side project, each day only had 24 hours, some of which were taken up by things other than pretalx. This led to having to prioritise features that a) I wanted to work on, b) I needed to work on, or c) that I was paid to work on. And since I didn’t work on it and nobody else did, either – well, features don’t get implemented by osmosis, yet.)

Happy to review PRs that implement a pared-down version of the feature.
